src/test/netbase_tests.cpp

```diff
@@ -443,4 +447,105 @@ BOOST_AUTO_TEST_CASE(netbase_dont_resolve_strings_with_embedded_nul_characters)
     BOOST_CHECK(!LookupSubNet(std::string("5wyqrzbvrdsumnok.onion\0example.com\0", 35), ret));
 }
 
+// Since CNetAddr (un)ser is tested separately in net_tests.cpp here we only
+// try a few edge cases for port, service flags and time.
+
+static const std::vector<CAddress> fixture_addresses({
+    CAddress(
+        CService(CNetAddr(in6addr_loopback), 0 /* port */),
+        NODE_NONE,
+        0x4966bc61U /* Fri Jan  9 02:54:25 UTC 2009 */
+    ),
+    CAddress(
+        CService(CNetAddr(in6addr_loopback), 0x00f1 /* port */),
+        NODE_NETWORK,
+        0x83766279U /* Tue Nov 22 11:22:33 UTC 2039 */
+    ),
+    CAddress(
+        CService(CNetAddr(in6addr_loopback), 0xf1f2 /* port */),
```

Comment:
```
+ make -C src --jobs=1 check-security V=1
make: Entering directory '/bitcoin/distsrc-x86_64-linux-gnu/src'
Checking binary security...
READELF=/gnu/store/pzq9s5ldl90h72p26jkdakyccdg9ib3k-profile/bin/x86_64-linux-gnu-readelf OBJDUMP=x86_64-linux-gnu-objdump OTOOL= /gnu/store/pzq9s5ldl90h72p26jkdakyccdg9ib3k-profile/bin/python3.7 ../contrib/devtools/security-check.py bitcoind  bitcoin-cli bitcoin-tx bitcoin-wallet test/test_bitcoin bench/bench_bitcoin qt/bitcoin-qt  
make: Leaving directory '/bitcoin/distsrc-x86_64-linux-gnu/src'
+ case "$HOST" in
+ make -C src --jobs=1 check-symbols V=1
make: Entering directory '/bitcoin/distsrc-x86_64-linux-gnu/src'
Checking glibc back compat...
READELF=/gnu/store/pzq9s5ldl90h72p26jkdakyccdg9ib3k-profile/bin/x86_64-linux-gnu-readelf CPPFILT=/gnu/store/pzq9s5ldl90h72p26jkdakyccdg9ib3k-profile/bin/x86_64-linux-gnu-c++filt /gnu/store/pzq9s5ldl90h72p26jkdakyccdg9ib3k-profile/bin/python3.7 ../contrib/devtools/symbol-check.py bitcoind  bitcoin-cli bitcoin-tx bitcoin-wallet test/test_bitcoin bench/bench_bitcoin qt/bitcoin-qt  
test/test_bitcoin: export of symbol in6addr_loopback not allowed
test/test_bitcoin: failed EXPORTED_SYMBOLS
make: *** [Makefile:20952: check-symbols] Error 1
make: Leaving directory '/bitcoin/distsrc-x86_64-linux-gnu/src'

Reply:
This breaks the guix/gitian build. cc @fanquake

Reply:
The patch below may fix it, but I can't confirm because on my machine `contrib/devtools/symbol-check.py` is happy with `src/test/test_bitcoin` which contains:

```sh
nm src/test/test_bitcoin |grep in6addr_loopback
                 U in6addr_loopback
```

<details>
<summary>possible workaround symbol-check.py in6addr_loopback</summary>

```diff
diff --git i/src/test/netbase_tests.cpp w/src/test/netbase_tests.cpp
index eb0d95a37..f473c092a 100644
--- i/src/test/netbase_tests.cpp
+++ w/src/test/netbase_tests.cpp
@@ -447,25 +447,27 @@ BOOST_AUTO_TEST_CASE(netbase_dont_resolve_strings_with_embedded_nul_characters)
     BOOST_CHECK(!LookupSubNet(std::string("5wyqrzbvrdsumnok.onion\0example.com\0", 35), ret));
 }
 
 // Since CNetAddr (un)ser is tested separately in net_tests.cpp here we only
 // try a few edge cases for port, service flags and time.
 
+static const struct in6_addr in6addr_loopback_ = IN6ADDR_LOOPBACK_INIT;
+
 static const std::vector<CAddress> fixture_addresses({
     CAddress(
-        CService(CNetAddr(in6addr_loopback), 0 /* port */),
+        CService(CNetAddr(in6addr_loopback_), 0 /* port */),
         NODE_NONE,
         0x4966bc61U /* Fri Jan  9 02:54:25 UTC 2009 */
     ),
     CAddress(
-        CService(CNetAddr(in6addr_loopback), 0x00f1 /* port */),
+        CService(CNetAddr(in6addr_loopback_), 0x00f1 /* port */),
         NODE_NETWORK,
         0x83766279U /* Tue Nov 22 11:22:33 UTC 2039 */
     ),
     CAddress(
-        CService(CNetAddr(in6addr_loopback), 0xf1f2 /* port */),
+        CService(CNetAddr(in6addr_loopback_), 0xf1f2 /* port */),
         static_cast<ServiceFlags>(NODE_WITNESS | NODE_COMPACT_FILTERS | NODE_NETWORK_LIMITED),
         0xffffffffU /* Sun Feb  7 06:28:15 UTC 2106 */
     )
 });
 
 // fixture_addresses should equal to this when serialized in V1 format.
```

Reply:
See #19954.